packages/rich-text-editor/src/vaadin-rich-text-editor.js

```diff
@@ -397,9 +367,56 @@ class RichTextEditor extends RichTextEditorMixin(
       ></vaadin-rich-text-editor-popup>
 
       <slot name="tooltip"></slot>
+
+      <slot name="link-dialog"></slot>
     `;
   }
 
+  /**
+   * Override update to render slotted link dialog into light DOM after rendering shadow DOM.
+   * @param changedProperties
+   * @protected
+   */
+  update(changedProperties) {
+    super.update(changedProperties);
+
+    this.__renderLinkDialog();
+  }
+
+  /** @private */
+  __renderLinkDialog() {
+    render(
+      html`
+        <vaadin-confirm-dialog
+          slot="link-dialog"
+          cancel-button-visible
+          .opened="${this._linkEditing}"
+          .header="${this.__effectiveI18n.linkDialogTitle}"
+          .rejectButtonVisible="${!!this._linkRange}"
+          @opened-changed="${this._onLinkEditingChanged}"
```

Comment:
I think we actually don't need to render slotted buttons? Maybe we could use this instead, it seems to work:

```suggestion
          .confirmText="${this.__effectiveI18n.ok}"
          .rejectText="${this.__effectiveI18n.remove}"
          .cancelText="${this.__effectiveI18n.cancel}"
          @confirm="${this._onLinkEditConfirm}"
          @cancel="${this._onLinkEditCancel}"
          @reject="${this._onLinkEditRemove}"
          @opened-changed="${this._onLinkEditingChanged}"
```